plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -146,12 +147,28 @@ public static boolean partitionMatchesPredicates(
             List<String> partitionValues,
             TupleDomain<HiveColumnHandle> constraintSummary)
     {
-        HivePartition partition = HivePartitionManager.parsePartition(
+        HivePartition partition = parsePartition(
                 tableName, hivePartitionName, partitionColumnHandles, partitionValues);
 
         return partitionMatches(partitionColumnHandles, constraintSummary, partition);
     }
 
+    public static HivePartition parsePartition(
+            SchemaTableName tableName,
+            String partitionName,
+            List<HiveColumnHandle> partitionColumns,
+            List<String> partitionValues)
+    {
+        ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builderWithExpectedSize(partitionColumns.size());
+        for (int i = 0; i < partitionColumns.size(); i++) {
+            HiveColumnHandle column = partitionColumns.get(i);
+            NullableValue parsedValue = parsePartitionValue(partitionName, partitionValues.get(i), column.getType());
+            builder.put(column, parsedValue);
+        }
+        Map<ColumnHandle, NullableValue> values = builder.buildOrThrow();
+        return new HivePartition(tableName, partitionName, values);
+    }
+
```

Comment:
Let's add a comment/javadoc here to show where this is copied from and the rationale for copying this, i.e. restrict changes to hudi-trino-module.